### PR TITLE
GRAL-826 Use form-data library for multipart content

### PR DESCRIPTION
**Pull request by eerootsus**
Fixes https://github.com/pipedrive/client-nodejs/issues/50

Using `form-data` library to correctly submit multipart content, including binary files. Thanks for @scottsbaldwin for raising the issue and working out the solution.

### CODE REVIEW

File: lib/rest.js [hunk 1]
```diff
@@ -99,7 +89,7 @@
 			rejectUnauthorized: process.env.PIPEDRIVE_API_HOST == 'api.pipedrive.com' ? true : false
 		};
 
-		log('Performing API request: ' + JSON.stringify(_.extend(requestOptions, { url: url }), null, 4));
+		log('Adding API request to queue: ' + JSON.stringify(_.extend(_.omit(requestOptions, this.options.multipart ? ['payload'] : []), { url: url }), null, 4));
```
Comment (tot-ra) [inline lib/rest.js]: What queue?

File: lib/rest.js [hunk 2]
```diff
@@ -47,31 +46,22 @@
...
 				} else {
-					payload += ' name="'+inputName.replace(/"/g, '\"')+'"';
+					form.append(inputName, value);
 				}
-
-				payload += '\r\n\r\n'+value+'\r\n';
 			});
-			payload += '--'+boundary+'--';
 
+            headers = form.getHeaders(headers);
```
Comment (tot-ra) [inline lib/rest.js]: Indentation

File: lib/rest.js (hunk 1 above)
Comment (eerootsus) [reply lib/rest.js]: @tot-ra The request is pushed into `requestQueue` two lines down. So if rate limiting kicks in the request is actually not immediately made.

File: lib/rest.js (hunk 2 above)
Comment (eerootsus) [reply lib/rest.js]: @tot-ra My bad, this project uses tabs and I didn't notice that.

File: lib/rest.js [hunk 3]
```diff
@@ -47,31 +46,22 @@
...
-			headers['content-type'] = 'multipart/form-data; boundary='+boundary;
-			payload = '';
+			var form = new FormData();
 
 			_.each(this.options.data, function(value, inputName) {
-				payload += '--'+boundary+'\r\n'+
-				'Content-Disposition: form-data;';
-				if(inputName == 'file_path' && fs.existsSync(value)) {
-					var parts = value.split(/\/|\\/);
+				if(inputName === 'file_path' && fs.existsSync(value)) {
```
Comment (ziimk) [inline lib/rest.js]: Please add space after `if` for consistency

**Pull request merged**